src/Aspire.Cli/Commands/PublishCommandBase.cs

```diff
@@ -516,47 +538,90 @@ private async Task HandlePromptActivityAsync(PublishingActivity activity, IAppHo
             throw new InvalidOperationException("Prompt provided without input data.");
         }
 
-        // Check for validation errors. If there are errors then this isn't the first time the user has been prompted.
-        var hasValidationErrors = inputs.Any(input => input.ValidationErrors is { Count: > 0 });
+        // If any inputs are still loading then display a loading status until they're available.
+        if (inputs.Any(input => input.Loading))
```

Comment:
I couldn't get the end-to-end working for this when I rewrote the subscription prompting in PublishModeProvisioningContextProvider. It seems like we never send a request back to the AppHost to invoke the dynamic input callback so we call directly into `HandleSelectInputAsync` with an empty choice list and it falls back to string-based prompting?

We can probably test this by unifying the implementations for prompting in `PublishModeProvisioningContextProvider` and `RunModeProvisioningContextProvider` to use the dynamic resolver and validate it works in both dashboard and CLI.